### ISSUE TITLE
Highlight values in the Object Inspector on double-click

### ISSUE THREAD

**Issue by deanljohnson** [AdamsLair/duality#259]
In almost any text interface, such as the editor used to write this, double clicking a word or number will highlight that word or number, allowing you to quickly replace it. However, this currently does not occur in the Object Inspector when you double click a value, which makes the inspector text fields behave differently than in most other UI's.

This is by no means a big deal, but I think it would be a nice usability feature.

**Comment by ilexp** [comment]
Will investigate this. Thanks for reporting!

**Comment by deanljohnson** [comment]
I dived into the source to try and see what I could find and I had to venture into your custom Winforms controls repo. The PropertyEditor class has a virtual OnMouseDoubleClick method without an implementation, and none of it's derivatives implement the method. The method is called by PropertyGrid however, so it seems like controls just need to override that method.

I'm looking into it some more now, but I'm not sure if I will be able to do anything about it. I can get you some information at least.

**Comment by ilexp** [comment]
> I'm looking into it some more now, but I'm not sure if I will be able to do anything about it, or even if you accept changes to your WinForms repo. I can get you some information at least.

I don't currently accept PRs to the WinForms repo (not as OSS-friendly as Duality) and it will probably more efficient in this case if I'll just take a quick look myself. This shouldn't be too big. Still, your first overview is very much appreciated! 

Implementing OnMouseDoubleClick in the text-based single value property editor(s) with the appropriate behavior should do the trick, with maybe a bit of functionality added to the editor template implementations as well, so the functionality will be used in more specialized / complex editors as well.

**Comment by deanljohnson** [comment]
Alright well I hope I was able to help. I got something working in most cases in the Winforms TestApp, but it only works on NumericPropertyEditor controls. NumericPropertyEditor overrides OnDoubleClick, calls an OnDoubleClick function on NumericEditorTemplate, who calls OnDoubleClick on StringEditorTemplate. In StringEditorTemplate.OnDoubleClick() just sets this.selectionLength to the length of this.text and calls UpdateScroll and EmitInvalidate. Works perfectly for numeric fields, but string fields, such as those labeled SomeString and SomeString2 aren't updated yet. Probably just need to pass the OnDoubleClick call around.
